StreamAPI/src/Demo.java

```diff
@@ -0,0 +1,273 @@
+        List<String> list3 = Arrays.asList(ba7.getIBANN(), ba8.getIBANN(), ba9.getIBANN(), ba10.getIBANN());
+
+        Person8Exercise pers1 = new Person8Exercise("Anna", list1);
+        Person8Exercise pers2 = new Person8Exercise("Teo", list2);
+        Person8Exercise pers3 = new Person8Exercise("Lia", list3);
+
+        List<Person8Exercise> personList = Arrays.asList(pers1, pers2, pers3);
+
+        List<String> res = personList.stream()
+                .flatMap(l -> l.getBanAccounts().stream())
+                .map(Demo::transformString)
+                //.map(iban -> transformString(iban))
+                .collect(Collectors.toList());
+
+        res.forEach(System.out::println);
+
+        System.out.println("-----------9------------------");
+        /*
+        Для списка persons посчитать общий возраст тех, кому больше 17 лет.
+        Подсказка - reduce()
+        */
+
+        Person p_1 = new Person("Anna", 40, a1);
+        Person p_2 = new Person("David", 40, a2);
+        Person p_3 = new Person("Robert", 25, a3);
+        Person p_4 = new Person("Lia", 16, a4);
+
+        List<Person> listP = Arrays.asList(p_1, p_2, p_3, p_4);
+
+        int sum = listP.stream().map(Person::getAge)
+        //int sum = listP.stream().map(a -> a.getAge())
+                .filter(p -> p > 17)
+                .reduce(Integer::sum)
+                //.reduce((acc, age) -> acc +age)
+                .get();
+        System.out.println(sum);
+
+        System.out.println("-----------10------------------");
+        /*
+        Написать функцию, которая для списка persons напечатает для тех, кто
+        старше 17 лет: In Germany <name1> and <name2> and …<nameN> are of legal
+        age. Подсказка - Collectors.joining();
+         */
+        List<Person> listPersons = Arrays.asList(p_1, p_2, p_3, p_4);
+
+        String sentence = listPersons.stream()
+                .filter(p -> p.getAge() > 17)
+                .map(Person::getName)
+                //.map(p -> p.getName())
+                .collect(Collectors.joining(" and ", "In Germany ", " are of legal age."));
+        System.out.println(sentence);
+
+
+    }
+
+    //============================Methods for Task 5=============================
+    public static String transformString (String b){
+        if(b == null) return null;
+
+        /* ---------------1 Method----------------
+        for (int i = 3; i < b.length(); i++) {
+            b = b.substring(0, i) + '*' + b.substring(i + 1);
+        }
+        return b;
+        */
+        /* -------------------2 Method ---------------
+        char[] chars = b.toCharArray();
+        for (int i = 3; i < b.length(); i++) {
+            chars[i] = '*';
+        }
+        String res = String.valueOf(chars);
+        return res;
+        */
+        // -------------------3 Method ---------------
+        StringBuilder res = new StringBuilder(b);
+        for (int i = 3; i < b.length(); i++) {
+            res.setCharAt(i,'*');
```

Comment:
If you know you need to replace 3 characters, why do you need a loop then?

Reply:
Sorry for confusion, I don't want to replace 3 characters, but starting from 3rd character till the end of string. :)

Reply:
Ok, then, it makes sense in case you do not know the length of the iban, it may vary in different countries a bit